fendermint/vm/interpreter/src/chain.rs

```diff
@@ -238,48 +246,31 @@ where
                         height: p.height as u64,
                         block_hash: p.block_hash,
                     };
-                    let msg = self
-                        .gateway_caller
-                        .commit_parent_finality_msg(finality.clone())?;
-                    let (state, ret) = self
-                        .inner
-                        .deliver(state, VerifiableMessage::NotVerify(msg))
-                        .await?;
-                    if ret.is_err() {
-                        return Err(anyhow!("cannot commit parent finality"));
-                    }
 
-                    // stash validator changes
+                    let (state, _, (prev_height, prev_finality)) = self
+                        .commit_finality(state, &provider, finality.clone())
+                        .await?;
+                    tracing::debug!("commit finality parsed: prev_height {prev_height}, prev_finality: {prev_finality:?}");
 
                     // error happens if we cannot get the validator set from ipc agent after retries
-                    let validator_changes = provider.validator_changes(p.height as u64).await?;
-                    let msg = self
-                        .gateway_caller
-                        .store_validator_changes_msg(validator_changes)?;
-                    let (state, ret) = self
-                        .inner
-                        .deliver(state, VerifiableMessage::NotVerify(msg))
+                    let validator_changes = provider
+                        .validator_changes_from(prev_height + 1, p.height as u64)
+                        .await?;
+                    let (state, _, _) = self
```

Comment:
```suggestion
                    let (state, _ret, ()) = self
```
To make it clear that `()` is returned, and if not it wouldn't compile, so it can't be ignored.